AirCasting/CoreData/ExternalSessionEntity+CoreDataProperties.swift

```diff
@@ -49,3 +66,89 @@ extension ExternalSessionEntity {
 extension ExternalSessionEntity : Identifiable {
 
 }
+
+extension ExternalSessionEntity {
+
+    var pm1Stream: MeasurementStreamEntity? {
+        let match = measurementStreams?.first(where: { (stream) -> Bool in
+            (stream as? MeasurementStreamEntity)?.sensorName == "AirBeam3-PM1" ||
+                (stream as? MeasurementStreamEntity)?.sensorName == "AirBeam2-PM1" ||
+                    (stream as? MeasurementStreamEntity)?.sensorName == "AirBeam1-PM1"
+        })
+        let pm1Stream = match as? MeasurementStreamEntity
+        return pm1Stream
+    }
+
+    var pm2Stream: MeasurementStreamEntity? {
+        let match = measurementStreams?.first(where: { (stream) -> Bool in
```

Comment:
we should extract the boy of this `var`s into a function that takes a string as a param because all of those vars are exactly the same except the string.

```Swift
var pm2Stream: MeasurementStreamEntity? { streamWithSensorName(containing: "PM2.5") }
```